quetz/tasks/mirror.py

```diff
@@ -168,6 +136,46 @@ def __getitem__(self, path):
         return cache_path
 
 
+def get_from_cache_or_download(
+    repository: RemoteRepository,
+    cache: LocalCache,
+    target: str,
+    config: Config,
+    exclude=["repodata.json", "current_repodata.json"],
+):
+    """Serve from cache or download if missing."""
+
+    _, filename = os.path.split(target)
+    skip_cache = filename in exclude
+
+    chunksize = 10000
+
+    def data_iter(f):
+        chunk = f.read(chunksize)
+        while chunk:
+            yield chunk
+            # Do stuff with byte.
+            chunk = f.read(chunksize)
+
+    if skip_cache:
+        remote_file = repository.open(target)
+        data_stream = remote_file.file
+
+        return StreamingResponse(data_iter(data_stream))
+
+    if target not in cache:
+        # copy from repository to cache
+        remote_file = repository.open(target)
+        data_stream = remote_file.file
+        cache.dump(target, data_stream)
+
+    if config.local_store_redirect_enabled:
```

Comment:
The `get_from_cache_or_download` function is called whatever the pkgstore. It only depends on the channel kind (proxy). Maybe using `config.local_store` isn't a good idea?

Should we use a different name?

Reply:
hmm, we should rather use the pkgstore interface here, shouldn't we?
I need to check what type LocalCache is.

Reply:
OK, I just checked and looks like the localcache implementation is a bit disjoint from our pkgstore interface that would otherwise be easily usable to generate URLs. 

I think it would be worthwhile to rewrite this part of the code to use the pkgstore interface (to be able to proxy-cache on S3 / Azure as well). In that sense, I still think it makes sense to use the variables from the local_store and say that we should rewrite the localcache to support the pkgstore.

Reply:
I just pushed some changes to remove the `LocalCache`. I copy the files to the package store instead.
I did some quick tests and it seems fine. I don't think there is an issue with the conda packages.
For the repodata.json and current_repodata.json, I saw those files are created (empty) when creating the proxy channel. There are also a repodata.json.gz that won't be updated (which could be an issue I guess).